chainstate/storage/src/lib.rs

```diff
@@ -141,6 +144,7 @@ pub trait BlockchainStorageWrite:
     fn add_block(&mut self, block: &Block) -> Result<()>;
 
     /// Remove block from the database
+    // FIXME: why is Id<Block> accepted by value, thought Id<GenBlock> above is accepted by ref?
```

Comment:
A long time ago Id was not Copy because it's 32-bytes, and then it was changed to be Copy based on arguments related to compiler optimizations. So both passing by value and reference are scattered around arbitrarily. Ideally they all should be fixed, but I can't care enough for that now.

One thing to keep in mind though that values passed to storage, that are considered large, should be passed by reference, because storage systems will perform a clone internally. There's no way to seemlessly move a value into a memory-map (as in the example of lmdb).